hoomd/mpcd/DryingDropletStreamingMethod.h

```diff
@@ -0,0 +1,311 @@
+        {
+        MPI_Allreduce(&N_bounced, &N_bounced_total, 1, MPI_UNSIGNED, MPI_SUM, m_exec_conf->getMPICommunicator());
+        MPI_Exscan(&N_bounced, &N_before, 1, MPI_UNSIGNED, MPI_SUM, m_exec_conf->getMPICommunicator());
+        }
+    #endif // ENABLE_MPI
+
+    /* if N_bounced_total <= N_evap, pick all the particles to delete
+     * else pick N_evap particles out of N_bounced_total
+     */
+    if (N_bounced_total <= N_evap)
+        {
+        //pick all bounced particles and remove all of them
+        m_picks.resize(N_bounced);
+        ArrayHandle<unsigned int> h_picks(m_picks, access_location::host, access_mode::overwrite);
+        std::iota(h_picks.data, h_picks.data + N_bounced, 0);
+        m_Npick = N_bounced;
+        //calculating density if it will get changed alot, print a warning
+        Scalar currentdensity = (m_mpcd_pdata->getNGlobal() - N_bounced_total)/((4.*M_PI/3.)*end_R*end_R*end_R);
+        if (std::fabs(currentdensity - m_density) > Scalar(0.1))
+            {
+            m_exec_conf->msg->warning() << "Solvent density changed to: " << currentdensity << std::endl;
+            }
+        }
+    else
+        {
+        // do the pick logic
+        makeAllPicks(timestep, N_evap , N_bounced_total);
+        
+        /*
+         * Select the picks that lie on my rank, with reindexing to local mark indexes.
+         * This is performed in a do loop to allow for resizing of the GPUVector.
+         * After a short time, the loop will be ignored.
+         */
+        const unsigned int max_pick_idx = N_before + N_bounced;
+        bool overflowed = false;
+        do
+            {
+            m_Npick = 0;
+            const unsigned int max_Npick = m_picks.getNumElements();
+
+                {
+                ArrayHandle<unsigned int> h_picks(m_picks, access_location::host, access_mode::overwrite);
+                for (unsigned int i=0; i < N_evap; ++i)
+                    {
+                    const unsigned int pick = m_all_picks[i];
+                    if (pick >= N_before && pick < max_pick_idx)
+                        {
+                        if (m_Npick < max_Npick)
+                            {
+                            h_picks.data[m_Npick] = pick - N_before;
+                            }
+                        ++m_Npick;
+                        }
+                    }
+                }
+
+            overflowed = (m_Npick > max_Npick);
+            if (overflowed)
+                {
+                m_picks.resize(m_Npick);
+                }
+
+            } while (overflowed);
+        }
+    
+    applyPicks();
+    //finally removing the picked particles
+    m_mpcd_pdata->removeParticles(m_removed,
+                                  m_bounced,
+                                  m_mask,
+                                  timestep);
+    }
+
+
+unsigned int DryingDropletStreamingMethod::calculateNumbounced()        
+    {
+    unsigned int N_bounced = 0;
+        {
```

Comment:
Because these are all scoped inside this function, you don't need to use the curly brackets here anymore. That was only necessary when this code was inside the main stream method. You can remove and deindent